### ISSUE TITLE
lint: add stringXbytes checker that detects redundant conversions between these two

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#427]
There are few special cases for string<->bytes conversions that can simplify the code.
For example, one may copy `string` to `[]byte` without conversion.

----

Detects redundant string<->[]byte conversions.

**Before:**
```go
v := []byte{'a', 'b', 'c'} // Only simple runes
copy(b, []byte(s))
```

**After:**
```go
v := []byte("abc")
copy(b, s)
```

----

Though we may also want to match this:
```go
v := string('x')
// =>
v := "x"
```
And that would require a different check name.
Or maybe we can do a separate checker like `stringXrune` that may cover even more string<->rune cases (can't give other examples right away).

**Comment by Quasilyte** [comment]
This one is not very complicated and can be a good one for first-time contributors.
Let's mark it appropriately and leave unassigned for a while.

**Comment by Quasilyte** [comment]
I've verified that `gometalinter --enable-all` does not detect these.
